crates/storage/src/pack.rs

```diff
@@ -79,6 +126,23 @@ impl<T> Pack<T> {
     pub fn as_inner_mut(pack: &mut Pack<T>) -> &mut T {
         &mut pack.inner
     }
+
+    /// Sets the key.
+    fn set_key(mut self, value: Option<Key>) -> Self {
+        self.key = value;
+        self
+    }
+}
+
+impl<T> Drop for Pack<T>
+where
+    T: PackedLayout,
+{
+    fn drop(&mut self) {
+        if let Some(key) = self.key {
+            clear_spread_root_opt::<T, _>(&key, || Some(&self.inner))
```

Comment:
Not sure it makes sense to use `clear_spread_root_opt` if the `&self.inner` isn't actually an optional.